lib/CTownHandler.h

```diff
@@ -253,6 +253,21 @@ class DLL_LINKAGE CTown
 
 	template <typename Handler> void serialize(Handler &h, const int version)
 	{
+		if(version < 777 && !h.saving)

```

Comment:
You should not check for `h.saving` in this case. Feel free to raise exception if saving with old version but do not just ignore. `h.saving` check in serialization should be used to initialize newly added fields when loading old save.